appeals/web/src/server/appeals/appeal-details/development-description/__tests__/development-description.test.js

```diff
@@ -0,0 +1,103 @@
+			nock('http://test/')
+				.get(`/appeals/${appealData.appealId}/appellant-cases/${appealData.appellantCaseId}`)
+				.reply(200, { appellantCaseDataIncompleteOutcome });
+			const response = await request.get(`${baseUrl}/development-description/change`);
+
+			const elementInnerHtml = parseHtml(response.text).innerHTML;
+
+			expect(response.statusCode).toEqual(200);
+			expect(elementInnerHtml).toMatchSnapshot();
+			expect(elementInnerHtml).toContain('Change development description</h1>');
+			expect(elementInnerHtml).toContain('Enter the original description of the development');
+		});
+	});
+
+	describe('/POST /change', () => {
+		it('should re-direct to appellant case when the data is valid', async () => {
+			const apiCall = nock('http://test/')
+				.patch(`/appeals/${appealData.appealId}/appellant-cases/${appealData.appellantCaseId}`)
+				.reply(200, {});
+			const validData = {
+				developmentDescription: 'Some valid text'
+			};
+
+			const response = await request
+				.post(`${baseUrl}/development-description/change`)
+				.send(validData);
+
+			expect(response.statusCode).toEqual(302);
+			expect(apiCall.isDone()).toBe(true);
+			expect(response.text).toBe(`Found. Redirecting to ${baseUrl}`);
+		});
+
+		it('should re-render the development description change page with an error if the field is empty', async () => {
+			const apiCall = nock('http://test/')
+				.patch(`/appeals/${appealData.appealId}/appellant-cases/${appealData.appellantCaseId}`)
+				.reply(200, {});
+			const appellantCaseDataCall = nock('http://test/')
+				.get(`/appeals/${appealData.appealId}/appellant-cases/${appealData.appellantCaseId}`)
+				.reply(200, { appellantCaseDataIncompleteOutcome });
+			const invalidData = {
+				developmentDescription: ''
+			};
+
+			const response = await request
+				.post(`${baseUrl}/development-description/change`)
+				.send(invalidData);
+
+			expect(response.statusCode).toEqual(200);
+			expect(apiCall.isDone()).toBe(false);
+			expect(appellantCaseDataCall.isDone()).toBe(true);
+
+			const elementInnerHtml = parseHtml(response.text).innerHTML;
+			expect(elementInnerHtml).toContain('Change development description</h1>');
+			expect(elementInnerHtml).toContain('Enter the original description of the development');
+			expect(elementInnerHtml).toContain('Development description cannot be empty</a>');
+		});
+
+		it('should re-render the development description change page with an error if the field is over 1000 char', async () => {
+			const apiCall = nock('http://test/')
+				.patch(`/appeals/${appealData.appealId}/appellant-cases/${appealData.appellantCaseId}`)
+				.reply(200, {});
+			const appellantCaseDataCall = nock('http://test/')
+				.get(`/appeals/${appealData.appealId}/appellant-cases/${appealData.appellantCaseId}`)
+				.reply(200, { appellantCaseDataIncompleteOutcome });
+
+			const invalidData = {
+				developmentDescription: `${'a'.repeat(1001)}` // Creating a 1001 char string.
```

Comment:
the comment is probably unnecessary here

Reply:
Wasn't sure if anyone had seen repeat before :)